lib/dice/src/cert.rs

```diff
@@ -0,0 +1,340 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{alias_cert_tmpl, deviceid_cert_tmpl, SerialNumber};
+use hubpack::SerializedSize;
+use salty::constants::{
+    PUBLICKEY_SERIALIZED_LENGTH, SIGNATURE_SERIALIZED_LENGTH,
+};
+use salty::signature::Keypair;
+use serde::{Deserialize, Serialize};
+use serde_big_array::BigArray;
+use unwrap_lite::UnwrapLite;
+
+#[derive(Debug)]
+pub enum CertError {
+    BadSig,
+    NoPubKey,
+    NoSig,
+    NoSignData,
+    TooSmall,
+    NotFound,
+    NoCn,
+}
+
+#[derive(Deserialize, Serialize, SerializedSize)]
+pub struct DeviceIdCert(
```

Comment:
This cert is self signed and so we generate it locally. When we add the manufacturing protocol we'll generate a CSR, send it out to the mfg system and get a cert back. We'll then write it to flash where it will be effectively read-only as you expect.